db/sql/cached_transaction.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2018 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudwan/gohan/db/pagination"
+	"github.com/cloudwan/gohan/db/transaction"
+	"github.com/cloudwan/gohan/schema"
+	// Import mysql lib
+	_ "github.com/go-sql-driver/mysql"
+	// Import go-sqlite3 lib
+	_ "github.com/mattn/go-sqlite3"
+	// Import go-fakedb lib
+	"github.com/cloudwan/gohan/metrics"
+	"github.com/mitchellh/hashstructure"
+	_ "github.com/nati/go-fakedb"
+)
+
+type CachedState struct {
+	list     []*schema.Resource
+	total    uint64
+	isLocked bool
+}
+
+type CachedTransaction struct {
+	TxInterface
+	QueryCache map[string]CachedState
+}
+
+func MakeCachedTransaction(transx TxInterface) TxInterface {
+	cachedTransaction := &CachedTransaction{transx, nil}
+	cachedTransaction.ClearCache()
+	return cachedTransaction
+}
+
+func (tx *CachedTransaction) Create(resource *schema.Resource) error {
+	return tx.CreateContext(context.Background(), resource)
+}
+
+func (tx *CachedTransaction) CreateContext(ctx context.Context, resource *schema.Resource) error {
+	tx.ClearCache()
```

Comment:
Since create affects only one table I don't think it's necessary to clear all cache.